### PR TITLE
Replace checked with cpa

### PR DESCRIPTION
**Pull request by maxtremblay**
Uniformize the implementation of checked index

### CODE REVIEW

File: crates/cubecl-core/src/runtime_tests/index.rs [hunk 1]
```diff
@@ -14,6 +14,9 @@ pub fn kernel_assign<F: Float>(output: &mut Array<F>) {
 
         // out of bounds read should be read as 0.
         output[1] = output[2];
+
+        // output[0] = F::cast_from(output.buffer_len());
+        // output[1] = F::cast_from(output.buffer_len());
```
Comment (nathanielsimard) [inline crates/cubecl-core/src/runtime_tests/index.rs:19]: dead code?

File: crates/cubecl-core/src/ir/operation.rs [hunk 2]
```diff
@@ -370,33 +370,39 @@ pub struct FmaOperator {
...
-            cpa!(scope, unchecked(out[lhs]) = rhs);
-        }));
+    cpa!(scope, item = unchecked(lhs[rhs]));
+    cpa!(scope, out = select(inside_bound, item, zero));
+}
+
+#[allow(missing_docs)]
+pub fn expand_checked_index_assign(scope: &mut Scope, lhs: Variable, rhs: Variable, out: Variable) {
+    let array_len = scope.create_local(Item::new(Elem::UInt(UIntKind::U32)));
+    let inside_bound = scope.create_local(Item::new(Elem::Bool));
```
Comment (nathanielsimard) [inline crates/cubecl-core/src/ir/operation.rs:394]: Could use local_binding here (or in the next PR)

File: crates/cubecl-wgpu/tests/main.rs [hunk 3]
```diff
@@ -43,7 +45,9 @@ pub fn kernel_sum(output: &mut Tensor<f32>) {
 pub fn plane_sum() {
     let kernel = KernelSum::<WgpuRuntime>::new(settings(4, 1), tensor());
     let expected = load_kernel_string!("plane_sum.wgsl");
-    assert_eq!(compile(kernel), expected);
+    let compiled = compile(kernel);
+    println!("{}", compiled);
```
Comment (nathanielsimard) [inline crates/cubecl-wgpu/tests/main.rs:49]: remove print

File: crates/cubecl-wgpu/tests/main.rs [hunk 4]
```diff
@@ -26,7 +26,9 @@ pub fn slice_assign_kernel(input: &Tensor<f32>, output: &mut Tensor<f32>) {
 pub fn slice_assign() {
     let kernel = SliceAssignKernel::<WgpuRuntime>::new(settings(1, 1), tensor(), tensor());
     let expected = load_kernel_string!("slice_assign.wgsl");
-    assert_eq!(compile(kernel), expected);
+    let compiled = compile(kernel);
+    println!("{}", compiled);
```
Comment (nathanielsimard) [inline crates/cubecl-wgpu/tests/main.rs:30]: remove print

File: crates/cubecl-core/src/ir/operation.rs (hunk 2 above)
Comment (maxtremblay) [reply crates/cubecl-core/src/ir/operation.rs:394]: Will be in the next PR since it leads to an error in this one.

**Review by nathanielsimard — APPROVED** [review]

**Pull request merged**